misc/json-to-record-converter/src/main/java/io/ballerina/jsonmapper/JsonToRecordMapper.java

```diff
@@ -146,6 +152,31 @@ public static JsonToRecordResponse convert(String jsonString, String recordName,
         return DiagnosticUtils.getDiagnosticResponse(diagnosticMessages, response);
     }
 
+    /**
+     * This method returns the Ballerina code for the provided JSON value or the diagnostics.
+     *
+     * @param jsonString JSON string of the JSON value to be converted to Ballerina record
+     * @param recordName Name of the generated record
+     * @param isRecordTypeDesc To denote final record, a record type descriptor (In line records)
+     * @param isClosed To denote whether the response record is closed or not
+     * @return {@link JsonToRecordResponse} Ballerina code block or the Diagnostics
+     */
+    public static JsonToRecordResponse convert(String jsonString, String recordName, boolean isRecordTypeDesc,
+                                               boolean isClosed, boolean forceFormatRecordTypeDesc) {
+        List<DiagnosticMessage> diagnosticMessages = new ArrayList<>();
+        JsonToRecordResponse response = convert(jsonString, recordName, isRecordTypeDesc, isClosed);
+        FormattingOptions formattingOptions = FormattingOptions.builder()
+                .setForceFormattingOptions(ForceFormattingOptions.builder()
+                        .setForceFormatRecordFields(forceFormatRecordTypeDesc).build()).build();
+        try {
+            response.setCodeBlock(Formatter.format(response.getCodeBlock(), formattingOptions));
+        } catch (FormatterException e) {
+            DiagnosticMessage message = DiagnosticMessage.jsonToRecordConverter102(null);
+            diagnosticMessages.add(message);
+        }
+        return DiagnosticUtils.getDiagnosticResponse(diagnosticMessages, response);
```

Comment:
Let's do the overloading the other way around. We can use this method inside the deprecated method instead right?